Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp

```diff
@@ -517,3 +517,108 @@ void webkit_cookie_manager_delete_all_cookies(WebKitCookieManager* manager)
     webkit_website_data_manager_clear(manager->priv->dataManager, WEBKIT_WEBSITE_DATA_COOKIES, 0, nullptr, nullptr, nullptr);
 }
 #endif
+
+/**
+ * webkit_cookie_manager_replace_cookies:
+ * @cookie_manager: a #WebKitCookieManager
+ * @cookies: (element-type SoupCookie) (transfer none): a #GList of #SoupCookie to be added
+ * @operation: a #WebKitCookieManagerOperation
+ * @cancellable: (allow-none): a #GCancellable or %NULL to ignore
+ * @callback: (scope async): a #GAsyncReadyCallback to call when the request is satisfied
+ * @user_data: the data to pass to callback function
+ *
+ * Asynchronously set a list of #SoupCookie to the underlying storage.
+ *
+ * When the operation is finished, @callback will be called. You can then call
+ * webkit_cookie_manager_replace_cookies_finish() to get the result of the operation.
+ *
+ * Since: 2.42
+ */
+void webkit_cookie_manager_replace_cookies(WebKitCookieManager* manager, GList* cookies, WebKitCookieManagerOperation operation, GCancellable* cancellable, GAsyncReadyCallback callback, gpointer userData)
+{
+    g_return_if_fail(WEBKIT_IS_COOKIE_MANAGER(manager));
+    g_return_if_fail(cookies);
+    g_return_if_fail(g_list_length(cookies));
```

Comment:
This check is redundant, please remove it: a non-null `GList` always has elements. Quoting [the documentation](https://docs.gtk.org/glib/type_func.List.length.html):

> To check whether the list is non-empty, it is faster to check list against NULL.